jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ExceptionUtil.java

```diff
@@ -76,6 +77,45 @@ public static <T extends Throwable> T as(Class<T> type, Throwable throwable) thr
         }
     }
 
+    /**
+     * Create a copy of a Throwable, discarding causes and suppressed exceptions.
+     * The copy will be of the same type if it has a constructor that takes a String
+     * message, otherwise it will be of the same supertype (e.g. IOException).
+     * @param failure The Throwable to copy
+     * @return A copy of the Throwable with no cause or suppressed exceptions.
+     */
+    public static Throwable copyOf(Throwable failure)
+    {
+        if (failure == null)
+            return null;
+        for (Constructor<?> constructor : failure.getClass().getDeclaredConstructors())
+        {
+            Class<?>[] params = constructor.getParameterTypes();
+            if (params.length == 1 && params[0] == String.class)
+            {
+                try
+                {
+                    if (Modifier.isPublic(constructor.getModifiers()))
+                        return (Throwable)constructor.newInstance(failure.getMessage());
+                }
+                catch (Throwable ignored)
+                {
+                }
+                break;
+            }
+        }
```

Comment:
I would rather try to find the `<ExceptionClass>(Throwable)` constructor first and failing that the parameterless constructor, and pass the original failure as cause or using `initCause()` on the new exception.
The `ISE` is thrown for suppression loops, not for cause loops.

Use `failure.getClass().getConstructor(Throwable.class)`, which returns only the `public` constructor with that signature.

Reply:
The problem with looking for a specific constructor with `getConstructor` is that it throws a `NoSuchMethod` exception if there is no such constructor.    Internally getConstructor(types...) loops through the constructors, so there is no efficiency difference.

We do not want to use pass the original failure as a cause, as the whole point is to avoid exception loops.  As the javadoc of this method says: without cause or suppressed exceptions.

Reply:
The problem is that we do not want to create two exceptions and two stack traces for every Throwable that might have some special constructor.